R/simulate_data.R

```diff
@@ -0,0 +1,611 @@
+#' `pars_t$prob_post_ice1_dropout`.
+#' - Simulate an additional uninformative study drop-out with probabilities `pars_c$prob_dropout`
+#' and `pars_t$prob_dropout` at each visit.
+#'   The simulated time of drop-out is the subject's first visit which is affected by
+#' drop-out and data from this visit and all subsequent visits are consequently set to missing.
+#'   In addition, in case the subject is still on treatment at the subject's (first)
+#' visit affected by drop-out (i.e. if this occurs prior to ICE1),
+#'   then dropout also triggers discontinuation of study drug and a corresponding ICE2
+#' (study treatment discontinuation due to NSDCR reasons) is generated.
+#' - Adjust trajectories after ICE according to the given assumption expressed with
+#' the `post_ice_traj` argument.
+#' - Simulate additional intermittent missing outcome data as per arguments `pars_c$prob_miss`
+#' and `pars_t$prob_miss`.
+#'
+#' The probability of the ICE after each visit is modeled according to the following
+#' logistic regression model:
+#' `~ 1 + I(visit == 0) + ... + I(visit == n_visits-1) + I((x-alpha))` where:
+#' - `n_visits` is the number of visits (including baseline).
+#' - `alpha` is the baseline outcome mean.
+#' The term `I((x-alpha))` specifies the dependency of the probability of the ICE on
+#' the current outcome value.
+#' The corresponding regression coefficients of the logistic model are defined as follows:
+#' The intercept is set to 0, the coefficients corresponding to discontinuation after
+#' each visit for a subject with outcome equal to
+#' the mean at baseline are set according to parameters `pars_c$prob_ice1` (`pars_t$prob_ice1`),
+#' and the regression coefficient associated with the covariate `I((x-alpha))` is set
+#' to `log(pars_c$or_outcome_ice)` (`log(pars_t$or_outcome_ice)`).
+#'
+#' Please note that the baseline outcome cannot be missing nor be affected by any ICEs.
+#'
+#' @returns A `data.frame` containing the simulated data. It includes the following variables:
+#' - `id`: Factor variable that specifies the id of each subject.
+#' - `visit`: Factor variable that specifies the visit of each assessment. Visit `0` denotes
+#' the baseline visit.
+#' - `group`: Factor variable that specifies which treatment group each subject belongs to.
+#' - `outcome_bl`: Numeric variable that specifies the baseline outcome.
+#' - `outcome_noICE`: Numeric variable that specifies the longitudinal outcome assuming
+#' no ICEs.
+#' - `ind_ice1`: Binary variable that takes value `1` if the corresponding visit is
+#' affected by ICE1 and `0` otherwise.
+#' - `dropout_ice1`: Binary variable that takes value `1` if the corresponding visit is
+#' affected by the drop-out following ICE1 and `0` otherwise.
+#' - `dropout2`: Binary variable that takes value `1` if the corresponding visit is affected
+#' by the uninformative study drop-out.
+#' - `ind_ice2`: Binary variable that takes value `1` if the corresponding visit is affected
+#' by ICE2.
```

Comment:
Sorry I'm confused by these two variables, shouldn't they always be the same? From the description in the `set_simul_pars` function my understanding was that ICE2's are the cause of non-informative dropout ?

Reply:
Good point. You are right.

Reply:
I will only keep ind_ice2

Reply:
ok, as I mentioned you in the chat these two are indeed not the same thing. ice2 happens if dropout2 happens and the patient hadn't ice1 before. In case that ice1 happened at visit 2, dropout1 did not happen and dropout2 happened at visit 3, ice2 does not happen (since the patient has already discontinued from treatment) but there are some missing values due to dropout2 without this being an ice. 

However we might simplify this a bit and make dropout2 and ice2 be the same (according to the definition of ice2). That is, ice2 happens as it currently happens, but we avoid additional missing values after ice1 due to dropout2.
@wolbersm what do you think? Would you keep it as originally designed or would you make this little simplification?

Reply:
@nociale 
Thanks a lot for this. 
I thought about this again and, to me, it would be better so simulate ICE2 (NSDRC discontinuations) slightly differently. 
My proposal is to omit argument prob_dropout altogether and have instead the following arguments for ICE2 (NSDRC discontinuations):
- drop_ice2 [constant visit-wise probability of ICE2] 
- prob_post_ice2_dropout
I.e. ICE2 would be handled essentially in the same way as for ICE1 except that it is simpler in the sense that we would not include an argument or_outcome_ice2. 

If for a subject, both ICE1 and ICE2 are simulated to occur, then only the earlier of them counts. In case both ICEs are simulated to occur at the same time, I guess we can decide that we count is as ICE1.

Apologies for late suggestion! If this is complicated to change or you don't like the solutation, please also let me know.